pilot/pkg/bootstrap/server.go

```diff
@@ -497,35 +497,58 @@ func (s *Server) initMCPConfigController(args *PilotArgs) error {
 
 	for _, configSource := range s.mesh.ConfigSources {
 		securityOption := grpc.WithInsecure()
-		if configSource.TlsSettings != nil && configSource.TlsSettings.Mode == istio_networking_v1alpha3.TLSSettings_ISTIO_MUTUAL {
-			requiredFiles := []string{
-				path.Join(model.AuthCertsPath, model.RootCertFilename),
-				path.Join(model.AuthCertsPath, model.CertChainFilename),
-				path.Join(model.AuthCertsPath, model.KeyFilename),
+		if configSource.TlsSettings != nil &&
+			configSource.TlsSettings.Mode != istio_networking_v1alpha3.TLSSettings_DISABLE {
+			var credentialOption *creds.Options
+			switch configSource.TlsSettings.Mode {
+			case istio_networking_v1alpha3.TLSSettings_SIMPLE:
```

Comment:
Note that Galley doesn't support simple TLS.

Reply:
We simply support this when users have their own server.